test/swar/BasicOperations.cpp

```diff
@@ -26,6 +26,15 @@ using S32_32 = SWAR<32, uint32_t>;
 
 using S64_64 = SWAR<64, uint64_t>;
 
+static_assert(SWAR<8, u32>::fromLaneLiterals({0, 0, 0, 0}).value() == 0);
+static_assert(SWAR<8, u32>::fromLaneLiterals({0, 0, 0, 1}).value() == 1);
+static_assert(SWAR<8, u32>::fromLaneLiterals({8, 3, 2, 1}).value() == 0x08'03'02'01);
+static_assert(SWAR<8, u32>::fromLaneLiterals({42, 42, 42, 42}).value() == 0x2A'2A'2A'2A);
+
+static_assert(SWAR<4, u32>::fromLaneLiterals({0, 0, 0, 0, 0, 0, 0, 0}).value() == 0);
+static_assert(SWAR<4, u32>::fromLaneLiterals({0, 0, 0, 0, 0, 0, 0, 1}).value() == 1);
+static_assert(SWAR<4, u16>::fromLaneLiterals({8, 3, 2, 1}).value() == 0b1000'0011'0010'0001);
```

Comment:
In practice, people are going to set up something like S4_32::FLL({1,2,3,4})?

Reply:
at least i think this makes it more fun and easy to write tests and to reason about correctness of the code.